planet/src/main/webapp/js/placesItinerary.js

```diff
@@ -38,39 +41,75 @@ function initAutocomplete() {
     let options = {
         types: ['geocode']
     };
-    autocompleteStart = new google.maps.places.Autocomplete(startAddress,options); 
+    autocompleteStart = new google.maps.places.Autocomplete(startAddress,options);
+    autocompleteStart.setFields(['address_component']);
+    autocompleteStart.addListener('place_changed', updateStartingAddress);
+
     autocompleteEvent = new google.maps.places.Autocomplete(eventAddress,options);
 }
 
+function updateStartingAddress() {
+    const startAddress = document.getElementById('starting-address');
+    sessionStorage.setItem('start', startAddress.value);
+}
+
 export function renderPlaces() {
     displayedPlaces = {};
+    allPlaces = [];
+    placeCount = 0;
     const userId = firebase.auth().currentUser.uid;
     const placesRef = database.ref('users/' + userId + '/places');
-    placesRef.once('value', (placesSnapshot) => {
+    placesRef.orderByValue().once('value', (placesSnapshot) => {
         const placesContainer = document.getElementById('places');
         placesContainer.innerHTML = '';
         const savedPlaces = placesSnapshot.val();
         for (const placeId in savedPlaces) {
-            // Make request with fields
-            let placeRequest = {
-                placeId: placeId,
-                fields: ['place_id','name','rating','formatted_address',
-                    'opening_hours','photos','url']
-            };
-            let service = new google.maps.places.PlacesService(map);
-            service.getDetails(placeRequest, renderPlaceDetailsCallback); 
+            allPlaces.push(placeId);
         }
+        makePlaceRequests();
     });
 }
 
+function makePlaceRequests() {
+    hideLoadMoreButton();
+    let service = new google.maps.places.PlacesService(map);
+    for (let i = placeCount; i < placeCount + 10 ; i++) {
+        if (i >= allPlaces.length) {
+            return;
+        }
+        let placeId = allPlaces[i];
+        // Make request with fields
+        let placeRequest = {
+            placeId: placeId,
+            fields: ['place_id','name','rating','formatted_address',
+                'opening_hours','photos','url']
+        };
+        service.getDetails(placeRequest, renderPlaceDetailsCallback);
+    } 
+    setTimeout(showLoadMoreButton, 5000);
+}
+
 function renderPlaceDetailsCallback(place, status) {
     if (status == google.maps.places.PlacesServiceStatus.OK) {
         const placesContainer = document.getElementById('places');
         const placeElement = createPlaceElement(place);
         placesContainer.appendChild(placeElement);
+        placeCount += 1;
     } 
 }
 
+function hideLoadMoreButton() {
```

Comment:
+1 for having separate functions for `hide` and `show`

Reply:
:D